src/main/java/com/fakng/fakngagrgtr/parser/ApiParser.java

```diff
@@ -5,30 +5,39 @@
 import org.springframework.stereotype.Component;
 import org.springframework.web.reactive.function.client.WebClient;
 import org.springframework.web.reactive.function.client.WebClient.ResponseSpec;
+import org.springframework.web.util.UriBuilder;
 import reactor.core.publisher.Mono;
 
+import java.net.URI;
+import java.util.function.Function;
+
 @Component
 @Slf4j
 public abstract class ApiParser extends Parser {
 
     protected final WebClient webClient;
 
+    protected static WebClient createWebClient(String url) {
```

Comment:
Why do you need this method?

Reply:
> for building a URL on the fly

In further there can be need to change params on the fly and prevent from parsing and rebuilding url in weirdo way. Just use provided components to compose what you need

Reply:
Is it very expensive to do? I don't know in a lack of experience

Reply:
There should be way to get proper `uriBuilder` without creation `webClient`

Reply:
I don't think that 7 lines for building it in the code is better than 1 line of not building it there

Besides, functions as params don't make code more readable - that's for sure

Moreover, if you build a URL, then you should build the URL, not WebClient

We will need to just update those params once to get max amount of vacancies from each request. The code is much more readable with one-line url string reading

Why do we complicate things? Any practical need?

Reply:
> There's a bean in ApplicationContext.

No, there isnt. Spring creates WebClient on each injection from WebClientFactory bean cause WebClient is not thread-safe